exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/convert/ConvertStringToBoolean.java

```diff
@@ -37,7 +37,11 @@ public void setString(final String value) {
       baseWriter.setNull();
     } else {
       try {
-        baseWriter.setInt(Boolean.parseBoolean(value) ? 1 : 0);
+        int intValue = Boolean.parseBoolean(value) ? 1 : 0;
```

Comment:
Consider using `BooleanType.get` instead, it will cover all cases with numeric and literal values. Though we might need to move `BooleanType` to common module so it will be visible in vector one.

https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/expr/BooleanType.java#L70